Dockerfile

```diff
@@ -1,19 +1,52 @@
-FROM golang:1.16-alpine
+FROM golang:1.18-alpine3.16 AS go-builder
 
 # Set up dependencies
-ENV PACKAGES curl make git libc-dev bash gcc linux-headers eudev-dev python3
+ENV PACKAGES curl make git libusb-dev libc-dev bash gcc linux-headers eudev-dev python3
+
+# Install ca-certificates
+RUN set -eux; apk add --no-cache ca-certificates build-base;
+
+# Install minimum necessary dependencies
+RUN apk add --no-cache $PACKAGES
+
+# See https://github.com/CosmWasm/wasmvm/releases
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0/libwasmvm_muslc.aarch64.a /lib/libwasmvm_muslc.aarch64.a
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0/libwasmvm_muslc.x86_64.a /lib/libwasmvm_muslc.x86_64.a
+RUN sha256sum /lib/libwasmvm_muslc.aarch64.a | grep 7d2239e9f25e96d0d4daba982ce92367aacf0cbd95d2facb8442268f2b1cc1fc
+RUN sha256sum /lib/libwasmvm_muslc.x86_64.a | grep f6282df732a13dec836cda1f399dd874b1e3163504dbd9607c6af915b2740479
+
+# Copy the library you want to the final location that will be found by the linker flag `-lwasmvm_muslc`
+RUN cp /lib/libwasmvm_muslc.$(uname -m).a /lib/libwasmvm_muslc.a
 
 # Set working directory for the build
 WORKDIR /usr/local/app
 
 # Add source files
 COPY . .
 
-# Install minimum necessary dependencies, build persistenceCore, remove packages
-RUN apk add --no-cache $PACKAGES && make install
+# Force it to use static lib (from above) not standard libgo_cosmwasm.so file
+RUN LEDGER_ENABLED=false BUILD_TAGS="muslc linkstatic" make build
+RUN echo "Ensuring binary is statically linked ..." \
+  && (file /usr/local/app/build/persistenceCore | grep "statically linked")
 
-# Install ca-certificates
-RUN apk add --update ca-certificates
+FROM alpine:3.16
+
+COPY --from=go-builder /usr/local/app/build/persistenceCore /usr/bin/persistenceCore
+
+COPY contrib/local/ /opt/
+RUN chmod +x /opt/*.sh
+
+# Set up dependencies
+ENV PACKAGES curl make bash jq
+ENV CHAIN_BIN /usr/bin/persistenceCore
+
+# Install minimum necessary dependencies
+RUN apk add --no-cache $PACKAGES
+
+WORKDIR /opt
+
+# rest server, p2p, rpc
+EXPOSE 1317 26656 26657
```

Comment:
longer term, would expose grpc 9090 too

Reply:
let me add it here itself then, might forget down the line